frame/pablo/src/lib.rs

```diff
@@ -670,6 +669,10 @@ pub mod pallet {
 		type AccountId = T::AccountId;
 		type PoolId = T::PoolId;
 
+		fn pools() -> Vec<Self::PoolId> {
```

Comment:
Isn't it excess memory allocation? I mean creation of `Vec` with all pools. Is't better to iterate over storage in `get_pool()` function and use Closure there? Or it's the dame stuff?!